src/combinatorics/simple_graph/basic.lean

```diff
@@ -480,6 +504,28 @@ set.to_finset_card _
   (univ : finset G.edge_set).card = G.edge_finset.card :=
 fintype.card_of_subtype G.edge_finset (mem_edge_finset _)
 
+section from_edge_finset
+
+variable (s : finset (sym2 V))
+
+/--
+`from_edge_finset` constructs a `simple_graph` from a finset of edges, without loops.
+-/
+def from_edge_finset : simple_graph V := from_edge_set (s : set (sym2 V))
```

Comment:
I would suggest not including this lemma since there's a coercion from `finset` to `set`. That's to say, the following would have worked as a definition, so you may as well write `from_edge_set s`.
```suggestion
def from_edge_finset : simple_graph V := from_edge_set s
```

Reply:
My motivation behind introducing these definitions/lemmas was doing induction on the `edge_finset` (and also decreasing induction, starting from a complete graph). I think finite graphs are quite widespread and it's useful to have a separate definition.

Reply:
I agree that finite graphs are important, but I believe having a separate definition is unnecessary for this. In my comments I've written an instance that gives that `from_edge_set s` has a finite edge set (so you can use `edge_finset` on it) given the fact that `s` is finite, which I believe should be sufficient for your needs.

Reply:
OK, I don't have a lot of experience with `set.finite`, and using `finset`'s directly is more comfortable for me. I believe you that it's OK to have only the `set` version.